x/oracle/client/common/proof/signature.go

```diff
@@ -64,34 +61,81 @@ func recoverETHAddress(msg, sig, signer []byte) ([]byte, uint8, error) {
 }
 
 // GetSignaturesAndPrefix returns a list of TMSignature from Tendermint signed header.
-func GetSignaturesAndPrefix(info *types.SignedHeader) ([]TMSignature, error) {
+func GetSignaturesAndPrefix(info *types.SignedHeader) ([]TMSignature, CommonEncodedVotePart, error) {
 	addrs := []string{}
 	mapAddrs := map[string]TMSignature{}
-	for i, vote := range info.Commit.Signatures {
+
+	prefix, err := protoio.MarshalDelimited(
+		&tmproto.CanonicalVote{
+			Type:   tmproto.SignedMsgType(info.Commit.Type()),
+			Height: info.Commit.Height,
+			Round:  int64(info.Commit.Round),
+		},
+	)
+	if err != nil {
+		return nil, CommonEncodedVotePart{}, err
+	}
+
+    // The first byte is the size of the CanonicalVote
+	prefix = prefix[1:21]
+	if info.Commit.Round == 0 {
```

Comment:
If round == 0, prefix size will be 21 or 12? In my understand if round is 0 encoded vote should skip that field?

Reply:
If round == 0 then the size of the prefix should be 11 
else 20 (11 + 9)

Reply:
Firstly, I assumed that the value was not zero (size = 20), so I will cut it out later if it is zero (size = 11).